### PR TITLE
FileReader bugfix

### PR DESCRIPTION
**Pull request by MStorm90**
An error accured when workplanes had no vectorblocks. This commit fixes this problem.

### CODE REVIEW

**Comment by SebastianDirks** [comment]
@MStorm90 please merge the current version of main into your branch first, there are merge conflicts

File: ReaderWriter/OVFReaderWriter/FileReaderWriter/OVFFileReader.cs [hunk 1]
```diff
@@ -125,17 +128,20 @@ public override async Task OpenJobAsync(string filename, IFileReaderWriterProgre
...
 
-                    await Task.Run(() =>
+                    var task = Task.Run(() =>
                     {
                         _readJobShell(jobLUTindex);
                         progress.IsFinished = false;
                         _workPlaneLUTs = new WorkPlaneLUT[_jobShell.NumWorkPlanes];
                         for (int i_plane = 0; i_plane < _jobShell.NumWorkPlanes; i_plane++)
                         {
+                            Debug.WriteLine(i_plane);
```
Comment (ax-meyer) [inline ReaderWriter/OVFReaderWriter/FileReaderWriter/OVFFileReader.cs:138]: Please remove debugging output before pushing to production.
Also remove the System.Diagnostics libary above if not needed anymore.

File: ReaderWriter/OVFReaderWriter/FileReaderWriter/OVFFileReader.cs [hunk 2]
```diff
@@ -212,20 +219,26 @@ public override async Task<Job> CacheJobToMemoryAsync()
 
                 _job = _jobShell.Clone();
                 _numberOfLayers = _job.NumWorkPlanes;
-                Task<WorkPlane>[] tasks = new Task<WorkPlane>[_numberOfLayers];
-                Parallel.For(0, _numberOfLayers, j =>
```
Comment (ax-meyer) [inline ReaderWriter/OVFReaderWriter/FileReaderWriter/OVFFileReader.cs:216]: Please restore parallel parsing functionality.

File: ReaderWriter/OVFReaderWriter/FileReaderWriter/OVFFileReader.cs [hunk 3]
```diff
@@ -288,28 +301,36 @@ private WorkPlane GetWorkPlaneShell(int i_workPlane, Stream stream)
                     stream = CreateLocalStream(i_workPlane);
                 }
                 WorkPlaneLUT wpLUT = _workPlaneLUTs[i_workPlane];
-                stream.Position = wpLUT.WorkPlaneShellPosition - GetStreamOffset(i_workPlane);
```
Comment (ax-meyer) [inline ReaderWriter/OVFReaderWriter/FileReaderWriter/OVFFileReader.cs:291]: Is there a reason to keep this simple calculation split up?

**Review by ax-meyer — COMMENTED** [review]
Please revert debugging stuff (debugging output, splitting up calculations over multiple lines) and restore parallel parsing of workplanes when caching the job.

**Pull request merged**